MobileCenter/MobileCenter/Internals/Storage/MSFileUtil.m

```diff
@@ -43,87 +43,86 @@ - (NSFileManager *)fileManager {
+    MSLogError([MSMobileCenter logTag], @"Error writing file %@: %@", file.fileURL, error.localizedDescription);
     return NO;
   }
 }
 
 + (BOOL)deleteFile:(MSFile *)file {
-  if (!file.filePath) {
+  if (!file.fileURL) {
     return NO;
   }
 
   NSError *error;
-  if ([self.fileManager removeItemAtPath:file.filePath error:&error]) {
-    MSLogVerbose([MSMobileCenter logTag], @"File %@: has been successfully deleted", file.filePath);
+  if ([self.fileManager removeItemAtURL:file.fileURL error:&error]) {
+    MSLogVerbose([MSMobileCenter logTag], @"File %@: has been successfully deleted", file.fileURL);
     return YES;
   } else {
-    MSLogError([MSMobileCenter logTag], @"Error deleting file %@: %@", file.filePath, error.localizedDescription);
+    MSLogError([MSMobileCenter logTag], @"Error deleting file %@: %@", file.fileURL, error.localizedDescription);
     return NO;
   }
 }
 
 + (nullable NSData *)dataForFile:(MSFile *)file {
-  if (!file.filePath) {
+  if (!file.fileURL) {
     return nil;
   }
 
   NSError *error;
-  NSData *data = [NSData dataWithContentsOfFile:file.filePath options:nil error:&error];
+  NSData *data = [NSData dataWithContentsOfURL:file.fileURL options:nil error:&error];
   if (error) {
-    MSLogError([MSMobileCenter logTag], @"Error writing file %@: %@", file.filePath, error.localizedDescription);
+    MSLogError([MSMobileCenter logTag], @"Error writing file %@: %@", file.fileURL, error.localizedDescription);
   } else {
-    MSLogVerbose([MSMobileCenter logTag], @"File %@: has been successfully written", file.filePath);
+    MSLogVerbose([MSMobileCenter logTag], @"File %@: has been successfully written", file.fileURL);
   }
   return data;
 }
 
-+ (nullable NSArray<MSFile *> *)filesForDirectory:(nullable NSString *)directoryPath
-                                 withFileExtension:(nullable NSString *)fileExtension {
++ (nullable NSArray<MSFile *> *)filesForDirectory:(nullable NSURL *)directoryURL
+                                withFileExtension:(nullable NSString *)fileExtension {
   NSFileManager *fileManager = [NSFileManager defaultManager];
+  NSError *error = nil;
 
   // Check validity.
-  if (!directoryPath || !fileExtension) {
+  if (!directoryURL || !fileExtension) {
     return nil;
   }
-  NSString * _Nonnull path = ( NSString* _Nonnull ) directoryPath;
 
   // Check file existing
-  if (![fileManager fileExistsAtPath:path]) {
+  if (![directoryURL checkResourceIsReachableAndReturnError:&error]) {
     return nil;
   }
 
   NSMutableArray<MSFile *> *files;
-  NSError *error;
-  NSArray *allFiles = [fileManager contentsOfDirectoryAtPath:path error:&error];
+  NSArray *allFiles = [fileManager contentsOfDirectoryAtURL:(NSURL * _Nonnull)directoryURL
+                                 includingPropertiesForKeys:@[NSURLCreationDateKey]
+                                                    options:NSDirectoryEnumerationSkipsHiddenFiles
```

Comment:
I would not skip hidden files to be more future proof.
@jaelim-ms @TroubleMakerBen Any though?